File: LeetCode-Solved-Problems/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l=0
        r=len(nums)-1
        while l<r:
            mid=(l+r)//2
            if nums[mid] > nums[len(nums)-1]:
                l = mid+1
            else:
                r=mid
        
        return nums[l];

File: LeetCode-Solved-Problems/0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        // if you do it linear it'll be too slow so, you need to do it
        // in log(n) using binary search
        int l=0,r=nums.size()-1,mid;
        while(l<r){
            mid=(l+r)/2;
            if(nums[mid]>nums[nums.size()-1])
                l=mid+1;
            else
                r=mid;
        }
        
        return nums[l];
    }
};